bookkeeper-server/src/main/java/org/apache/bookkeeper/client/EnsemblePlacementPolicy.java

```diff
@@ -408,4 +407,25 @@ default boolean isEnsembleAdheringToPlacementPolicy(List<BookieSocketAddress> en
             int ackQuorumSize) {
         return false;
     }
+
+    /**
+     * Result of a placement calculation against a placement policy.
+     */
+    class PlacementResult<T> {
+        private final T result;
+        private final boolean adhering;
+
+        public PlacementResult(T result, boolean adhering) {
```

Comment:
what about making the constructor private and making the class 'final' ?

We can open to extensions on the future if needed

Reply:
I can't make it private because LocalBookieEnsemblePlacementPolicy is not in the same package. I've made it final though.

Reply:
And what about making constructor private and adding a public static 'of' method to build instances?

Reply:
@sijie what about this comment of mine?

Reply:
I am fine with either approach.